tests/service/SyntaxTreeTests.fs

```diff
@@ -0,0 +1,170 @@
+            SynModuleSigDecl.HashDirective(hd, m)
+        | SynModuleSigDecl.NestedModule(moduleInfo, isRecursive, decls, range, trivia) ->
+            SynModuleSigDecl.NestedModule(moduleInfo, isRecursive, List.map mapSynModuleDeclSig decls, range, trivia)
+        | _ -> msdl
+
+    let mapSynModuleOrNamespaceSig (SynModuleOrNamespaceSig(longId, isRecursive, kind, decls, xmlDoc, attribs, ao, range, trivia)) =
+        SynModuleOrNamespaceSig(longId, isRecursive, kind, List.map mapSynModuleDeclSig decls, xmlDoc, attribs, ao, range, trivia)
+
+    match ast with
+    | ParsedInput.ImplFile(ParsedImplFileInput(fileName,
+                                               isScript,
+                                               qualifiedNameOfFile,
+                                               scopedPragmas,
+                                               hashDirectives,
+                                               contents,
+                                               flags,
+                                               trivia,
+                                               identifiers)) ->
+        ParsedImplFileInput(
+            fileName,
+            isScript,
+            qualifiedNameOfFile,
+            scopedPragmas,
+            List.map mapParsedHashDirective hashDirectives,
+            List.map mapSynModuleOrNamespace contents,
+            flags,
+            trivia,
+            identifiers
+        )
+        |> ParsedInput.ImplFile
+    | ParsedInput.SigFile(ParsedSigFileInput(fileName, qualifiedNameOfFile, scopedPragmas, hashDirectives, contents, trivia, identifiers)) ->
+        ParsedSigFileInput(
+            fileName,
+            qualifiedNameOfFile,
+            scopedPragmas,
+            List.map mapParsedHashDirective hashDirectives,
+            List.map mapSynModuleOrNamespaceSig contents,
+            trivia,
+            identifiers
+        )
+        |> ParsedInput.SigFile
+
+let parseSourceCode (name: string, code: string) =
+    let location = Path.Combine(RootDirectory, name).Replace("\\", "/")
+
+    let parseResults =
+        checker.ParseFile(
+            location,
+            SourceText.ofString code,
+            { FSharpParsingOptions.Default with
+                SourceFiles = [| location |] }
+        )
+        |> Async.RunImmediate
+
+    let tree = parseResults.ParseTree
+    let sourceDirectoryValue = $"{RootDirectory}/{FileInfo(location).Directory.Name}"
+    sanitizeAST sourceDirectoryValue tree
+
+/// Asserts the parsed untyped tree matches the expected baseline.
+///
+/// To update a baseline:
+///     CMD: set TEST_UPDATE_BSL=1 & dotnet test --filter "ParseFile"
+///     PowerShell: $env:TEST_UPDATE_BSL = "1" ; dotnet test --filter "ParseFile"
+///     Linux/macOS: export TEST_UPDATE_BSL=1 & dotnet test --filter "ParseFile"
+///
+/// Assuming your current directory is tests/FSharp.Compiler.Service.Tests
+[<TestCaseSource(nameof allTestCases)>]
+let ParseFile fileName =
```

Comment:
Is there a way to run a specific test? It's going to be crucial when debugging some cases (e.g. debugging helped me a lot when I worked on the parser in https://github.com/dotnet/fsharp/pull/13352/files#diff-72387be5568c5958f284243b583f8c1c8440e9ace7485dd9aae37a040c0d716dR989)

It would also be very handy to be able to dump tokens for a specific tests easily: it may really help when changing LexFilter, as in https://github.com/dotnet/fsharp/pull/13089.

@nojaf @vzarytovskii Would it be easy to have separate named tests, similar to what we have here? https://github.com/JetBrains/resharper-fsharp/blob/net231/ReSharper.FSharp/test/src/FSharp.Tests/Parsing/FSharpParserTest.fs

Reply:
Ouch, I'm a bit late 😞

Reply:
You can debug each individual file:
![image](https://user-images.githubusercontent.com/2621499/218077836-9dea11ad-3773-4ae0-b076-379a63a61418.png)